adapters/vidoomy/vidoomy.go

```diff
@@ -0,0 +1,192 @@
+package vidoomy
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+func (a adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errors []error
+
+	reqs := make([]*adapters.RequestData, 0, len(request.Imp))
+
+	for idx, imp := range request.Imp {
+		var bidderExt adapters.ExtImpBidder
+		err := json.Unmarshal(imp.Ext, &bidderExt)
+		if err != nil {
+			err = &errortypes.BadInput{
+				Message: fmt.Sprintf("imp #%d: ext.bidder not provided", idx),
+			}
+			errors = append(errors, err)
+			continue
+		}
+
+		var vidoomyExt openrtb_ext.ImpExtVidoomy
+		err = json.Unmarshal(bidderExt.Bidder, &vidoomyExt)
+		if err != nil {
+			err = &errortypes.BadInput{
+				Message: fmt.Sprintf("imp #%d: %s", idx, err.Error()),
+			}
+			errors = append(errors, err)
+			continue
+		}
```

Comment:
You may wish to consider using a different struct such that only one call to unmarshal is required. This will simplify the code here a bit. The data structure, which could be anonymous, would look something like:

```
type extBidder struct {
   vidoomyExt openrtb_ext.ImpExtVidoomy `json:"bidder"`
}
```

Reply:
Moved ext unmarshaling to a new function to simplify.